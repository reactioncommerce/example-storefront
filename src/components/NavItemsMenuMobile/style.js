import styled from "styled-components";

export const Navigation = styled.nav`
  svg {
    height: 25px;
    width: 25px;
  }
`;

export const BackButton = styled.button`
  height: 80px;
  padding: 0px 30px;
  svg {
    margin-left: -5px;
  }
`;

export const BackButtonText = styled.span`
  font-size: 16px;
  margin-left: 15px;
`;

export const LinkContainer = styled.div`
  margin-bottom: 15px;
  padding: 0px;

  a {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    height: 41px;
    font-size: 16px;
    color: #4f4d4d;
    outline: none;
    padding: 0px 30px;

    svg {
      margin-right: -5px;
    }
  }
`;
