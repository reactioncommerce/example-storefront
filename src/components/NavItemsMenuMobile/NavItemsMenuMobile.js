import React from "react";
import PropTypes from "prop-types";
import MenuRightIcon from "mdi-material-ui/MenuRight";
import MenuLeftIcon from "mdi-material-ui/MenuLeft";
import Link from "components/Link";
import * as styles from "./style";

const NavItemsMenuMobile = ({ handleClose }) => {
  const navItems = [
    { title: "Link", href: "/" },
    { title: "Categorias", href: "/" },
    { title: "Link", href: "/" },
    { title: "Link", href: "/" },
    { title: "Link", href: "/" }
  ];

  return (
    <styles.Navigation>
      <styles.BackButton id="close-menu" onClick={handleClose}>
        <MenuLeftIcon />
        <styles.BackButtonText>VOLTAR</styles.BackButtonText>
      </styles.BackButton>

      {navItems &&
        navItems.map((el, idx) => (
          <styles.LinkContainer key={idx}>
            <Link id={`page-link-${idx}`} route="/" onClick={handleClose}>
              Link
              <MenuRightIcon />
            </Link>
          </styles.LinkContainer>
        ))}
    </styles.Navigation>
  );
};

NavItemsMenuMobile.propTypes = {
  handleClose: PropTypes.func
};

export default NavItemsMenuMobile;
