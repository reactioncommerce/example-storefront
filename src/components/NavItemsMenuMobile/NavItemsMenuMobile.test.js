import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import NavItemsMenuMobile from "./NavItemsMenuMobile";

describe("NavItemsMenuMobile", () => {
  it("Should call handleClose function", () => {
    const mockCallBack = jest.fn();
    const component = shallow(<NavItemsMenuMobile handleClose={mockCallBack} />);
    component.find("#close-menu").simulate("click");
    component.find("#page-link-1").simulate("click");

    expect(mockCallBack.mock.calls.length).toEqual(2);
  });

  it("Should match the snapshot", () => {
    const tree = renderer.create(<NavItemsMenuMobile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
