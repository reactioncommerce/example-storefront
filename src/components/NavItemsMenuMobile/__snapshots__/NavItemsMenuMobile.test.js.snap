// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItemsMenuMobile Should match the snapshot 1`] = `
.c0 svg {
  height: 25px;
  width: 25px;
}

.c1 {
  height: 80px;
  padding: 0px 30px;
}

.c1 svg {
  margin-left: -5px;
}

.c2 {
  font-size: 16px;
  margin-left: 15px;
}

.c3 {
  margin-bottom: 15px;
  padding: 0px;
}

.c3 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 41px;
  font-size: 16px;
  color: #4f4d4d;
  outline: none;
  padding: 0px 30px;
}

.c3 a svg {
  margin-right: -5px;
}

<nav
  className="c0"
>
  <button
    className="c1"
    id="close-menu"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M14,7L9,12L14,17V7Z"
      />
    </svg>
    <span
      className="c2"
    >
      VOLTAR
    </span>
  </button>
  <div
    className="c3"
  >
    <a
      className="SkLink-anchor-10"
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Link
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,17L15,12L10,7V17Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c3"
  >
    <a
      className="SkLink-anchor-10"
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Link
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,17L15,12L10,7V17Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c3"
  >
    <a
      className="SkLink-anchor-10"
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Link
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,17L15,12L10,7V17Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c3"
  >
    <a
      className="SkLink-anchor-10"
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Link
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,17L15,12L10,7V17Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c3"
  >
    <a
      className="SkLink-anchor-10"
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Link
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,17L15,12L10,7V17Z"
        />
      </svg>
    </a>
  </div>
</nav>
`;
